libs/vkd3d/state.c

```diff
@@ -4758,6 +4829,13 @@ VkPipeline d3d12_pipeline_state_create_pipeline_variant(struct d3d12_pipeline_st
     *dynamic_state_flags = d3d12_graphics_pipeline_state_init_dynamic_state(state, &dynamic_create_info,
             dynamic_state_buffer, key);
 
+    if (!library_flags && graphics->library)
+    {
+        if (d3d12_pipeline_state_link_pipeline_variant(state, key, dsv_format,
+                vk_cache, *dynamic_state_flags, &vk_pipeline) == VK_SUCCESS)
+            return vk_pipeline;
```

Comment:
If this fails (due to e.g. identifiers), we might have some problems since we're falling back to a non-library path. Is that intentional? I'm not sure if we retain SPIR-V or DXBC at this stage.

It might be possible to synthesize this behavior by always returning FAIL_ON_COMPILE_REQUIRED in this call and making sure everything makes sense.

Reply:
We should be retaining DXBC since there are no changes to the `is_fully_dynamic` logic which governs this.